### ISSUE TITLE
Breakdance Cookies slow down server response

### ISSUE THREAD

**Issue by Nicscott01** [soflyy/breakdance-bugs#954]
I was noticing high server response times on many of my recent Breakdance sites. After much investigation and troubleshooting, I learned that the cookies (Privacy Settings in Breakdance) BD sets and reads are adding sometimes a few hundred milliseconds to the server response. This is mainly due to the inability of FastCGI to cache the page because it contains cookies in the header.

I don't think I've actually used any features of Breakdance that require these cookies (maybe instructing a Popup to stay closed after being shown). 

But I'm wondering if there's a better way to handle these cookies? 
Can they be set via JS and still serve the same purpose?

**Comment by thekendog** [comment]
Whoa, I assumed those cookies were already set with javascript, not PHP.  Just looked in the code though and yup, it's PHP.  That's going to be a huge issue as it will completely stop Nginx caching unless the `fastcgi_ignore_headers` option is set to include `Set-Cookie`

**Comment by thekendog** [comment]
Not sure how/if the Wordpress plugins like WP Rocket handle setcookie, but if they do continue to cache pages wouldn't that  make conditionals based on page/session views not work correctly?

**Comment by Nicscott01** [comment]
I'm not sure. I use [SpinupWP](https://spinupwp.com/?referral=xan58PN5BR) to host my sites, which uses ngnix/fastcgi to serve up pages without any caching plugins. I definitely have run into performance issues on each Breakdance site I've launched thus far (since March).

**Comment by thekendog** [comment]
@Nicscott01 yeah I use SpinupWP as well and they don't add the ignore header for Set-Cookie.  I checked last night.

**Comment by Nicscott01** [comment]
> @Nicscott01 yeah I use SpinupWP as well and they don't add the ignore header for Set-Cookie. I checked last night.

It was through their support chat that I realized Breakdance was using cookies because the fastcgi-cache header was returning MISS for everything!

**Comment by felipemarcos** [comment]
I'm investigating this, but a possible fix is to set the cookie client-side, is that right?

In the meanwhile, you could disable the cookie setting entirely by going to Breakdance > Settings > Privacy > Disable Page & Session Cookies.

**Comment by Nicscott01** [comment]
Has there been any movement on this?
